src/charm.py

```diff
@@ -50,6 +51,8 @@ def _is_ip_address_range(str_to_test):
 
 
 def _is_cidr(str_to_test):
+    if str_to_test.count("/") != 1:
+        return False
```

Comment:
Or test .+\/\d{1,2}$ ?

Reply:
nothing scarier here than using regex for ip address matching.  I'll let python check it.  See if you like my change here.  

i want to make sure it's a ip_network and NOT an ip_address.  The python library accepts a bare address to be processed as a "network" even when it's not in CIDR notation 

```python
assert ipaddress.ip_network("1.2.3.4") == ipaddress.ip_network("1.2.3.4/32")
```

HOWEVER, using ip_address totally raises a ValueError

````python
>>> ipaddress.ip_address("1.2.3.4/32")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/.pyenv/versions/3.12.3/lib/python3.12/ipaddress.py", line 54, in ip_address
    raise ValueError(f'{address!r} does not appear to be an IPv4 or IPv6 address')
ValueError: '1.2.3.4/32' does not appear to be an IPv4 or IPv6 address
```